src/main/kotlin/com/tschuchort/compiletesting/JavacUtils.kt

```diff
@@ -75,11 +75,16 @@ internal fun getJavacVersionString(javacCommand: String): String {
 
     val output = buffer.readUtf8()
 
-    return Regex("javac (.*)?[\\s\\S]*").matchEntire(output)?.destructured?.component1()
-        ?: throw IllegalStateException("Command '$javacCommand -version' did not print expected output. " +
-                "Output was: '$output'")
+    return parseVersionString(output, javacCommand)
 }
 
+internal fun parseVersionString(output: String, javacCommand: String) =
+    Regex("javac (.*)?[\\s\\S]*").find(output)?.destructured?.component1()
+        ?: throw IllegalStateException(
+            "Command '$javacCommand -version' did not print expected output. " +
+                    "Output was: '$output'"
+        )
+
```

Comment:
I don't like how `parseVersionString` has to know about `javacCommand` just for "logging" purposes. I think it would be better to let the `getJavacVersionString` function handle that:

```kotlin
internal fun getJavacVersionString(javacCommand: String): String {
    val javacProc = ProcessBuilder(listOf(javacCommand, "-version"))
        .redirectErrorStream(true)
        .start()
    val buffer = Buffer()
    javacProc.inputStream.copyTo(buffer.outputStream())
    javacProc.waitFor()

    val output = buffer.readUtf8()
    return parseVersionString(output) ?: throw IllegalStateException(
            "Command '$javacCommand -version' did not print expected output. " +
                    "Output was: '$output'"
        )
}

internal fun parseVersionString(output: String): String? =
    Regex("javac (.*)?[\\s\\S]*").find(output)?.destructured?.component1()
```

or

```kotlin
internal fun getJavacVersionString(javacCommand: String): String {
    val javacProc = ProcessBuilder(listOf(javacCommand, "-version"))
        .redirectErrorStream(true)
        .start()
    val buffer = Buffer()
    javacProc.inputStream.copyTo(buffer.outputStream())
    javacProc.waitFor()

    val output = buffer.readUtf8()
    return try {
        parseVersionString(output)
    }
    catch(e: Exception) {
        throw IllegalStateException("Unable to get javac version string from command '$javacCommand -version' ", e)
    }
}

internal fun parseVersionString(javacOutput: String): String =
    Regex("javac (.*)?[\\s\\S]*").find(javacOutput)?.destructured?.component1()
        ?: throw IllegalArgumentException("'$javacOutput' does not contain the expected regex pattern for javac version strings")
```

Reply:
Change the code according to the first suggestion. Try-catch seem to have a bad image in Kotlin :)
https://medium.com/@elizarov/kotlin-and-exceptions-8062f589d07